src/RequestProcessor.cs

```diff
@@ -281,6 +337,151 @@ private void BindOutputFromResult(InvocationResponse response, AzFunctionInfo fu
+                {
+                    // If a latest version was not found, error out.
+                    var errorMsg = string.Format(PowerShellWorkerStrings.CannotFindModuleVersion, moduleName, majorVersion);
+                    var argException = new ArgumentException(errorMsg);
+                    logger.Log(LogLevel.Error, errorMsg, argException, isUserLog: true);
+
+                    throw argException;
+                }
+
+                if (!DependencyManagementUtils.IsValidMajorVersion(majorVersion, latestVersion))
+                {
+                    // The requested major version is greater than the latest major supported version.
+                    var errorMsg = string.Format(PowerShellWorkerStrings.InvalidModuleMajorVersion, moduleName, majorVersion);
+                    var argException = new ArgumentException(errorMsg);
+                    logger.Log(LogLevel.Error, errorMsg, argException, isUserLog: true);
+
+                    throw argException;
+                }
+
+                // Before installing the module, check to see if it is already installed at the given path.
+                var moduleVersionFolderPath = Path.Combine(path, moduleName, latestVersion);
+                if (Directory.Exists(moduleVersionFolderPath))
+                {
+                    // The latest version is already installed, log and return.
+                    var logMsg = string.Format(PowerShellWorkerStrings.ModuleIsAlreadyInstalled, moduleName, latestVersion);
+                    logger.Log(LogLevel.Trace, logMsg, isUserLog: true);
+
+                    return;
+                }
+
+                // Save-Module is able to download versions of Az module side-by-size. However, the dependent modules get
+                // overwritten with the latest version. As a result, the previous Az version will not longer work.
+                // If this is the case, empty the Az directory before running Save-Module.
+                if (moduleName.Equals(AzModuleName, StringComparison.OrdinalIgnoreCase))
+                {
+                    var azModulePath = Path.Join(path, moduleName);
+                    DependencyManagementUtils.EmptyDirectory(azModulePath);
+                }
+
+                // Save the module to the given path
+                SaveModuleCommand(pwsh, logger, moduleName, latestVersion, path);
+            }
+        }
+
+        /// <summary>
+        /// Downloads a PowerShell module from the PSGallery to the given path.
+        /// </summary>
+        internal void SaveModuleCommand(
+            System.Management.Automation.PowerShell pwsh,
+            ILogger logger,
+            string moduleName,
+            string version,
+            string path)
+        {
+            Exception exception = null;
+
+            // If the destination path does not exist, create it.
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
```

Comment:
These can be abstracted using Package System.IO.Abstractions

https://github.com/System-IO-Abstractions/System.IO.Abstractions